workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs

```diff
@@ -48,22 +49,30 @@ internal class MonoBehaviourActivationManager
             foreach (var behaviour in gameObject.GetComponents<MonoBehaviour>())
             {
                 var behaviourType = behaviour.GetType();
-                var readerIds = injector.GetRequiredReaderComponentIds(behaviourType);
-                var writerIds = injector.GetRequiredWriterComponentIds(behaviourType);
-
-                if (readerIds.Count + writerIds.Count > 0)
+                if (injector.HasRequiredFields(behaviourType))
                 {
-                    if (readerIds.Count > 0)
+                    var componentReadRequirements = injector.GetComponentPresenceRequirements(behaviourType);
+                    var componentAuthRequirements = injector.GetComponentAuthorityRequirements(behaviourType);
+                    var readRequirementCount = componentAuthRequirements.Count;
+                    var authRequirementCount = componentAuthRequirements.Count;
+
+                    if (readRequirementCount == 0 && authRequirementCount == 0)
```

Comment:
Is this the case if a MonoBehaviour is not requiring anything at all? Is the idea here further that we pre-disable all monoBehaviours on a gameobject regardless of whether they require anything?

Reply:
This is the case of a MonoBehaviour requiring something, but only things that are always available (command senders / response handlers).

The idea is that for consistency we will manage preprocess and disable the MonoBehaviours if they Require anything, but in the case that all required fields are always available, we just immedieately enable them after all.